common/lib/xmodule/xmodule/services.py

```diff
@@ -0,0 +1,19 @@
+"""
+Module contains various XModule/XBlock services
```

Comment:
@sarina TL;DR `XBLOCK_SETTINGS` keys are strings; XBlocks can use their classname for settings bucket, or override it using `block_settings_key` attribute; tests exercise both ways.

The idea is the following. `XBLOCK_SETTINGS` contains dictionary (string) => (whatever, but probably dictionary). The key string is tied with corresponding XBlock in a following way:

* if XBlock has `block_settings_key` - the value of `block_settings_key`.
* XBlock class name otherwise.

I used `block_settings_key`, not `xblock_name` because it's not a name - it's a key to look up settings at.

Settings service is made content-agnostic in a sense that it does not care what actually lays in a bucket - it just returns to XBlock whatever happens to be in bucket it is assigned to.

So, the settings example I given earlier is a contains specific settings for mentoring XBlock that allow actually seeing settings service in action. On the other hand, tests capture that content-agnostic aspect of `SettingsService` by using various types of values for bucket content (string and dict for now, but I'm going to fix some issues there, so I'm adding more types).